Fix #380: Fresh dump of production database structure

kumy opened the pull request.

okainov commented on docker/mariadb/00_database-create-geokrety-db.sql:
```diff
@@ -526,42 +456,298 @@ CREATE TABLE `gk-waypointy` (
+('AT',	'Austria'),
+('AU',	'Australia'),
+('Australien',	'Australia'),
+('Austria',	'Australia'),
```
For some reason previous comments were not added to the lines as expected... I meant here

kumy replied:
> Austria

Fixed in database.

> Sounds like extra useless key :)

No context, I don't understand sorry.

> why not remove Polish comment and keep English?

Yep something to be done.

> I don't know the typical queries against this table, but as per amount of indexes looks like introducing some composite indexes may be helpful

Sounds like you're talking about `gk-ruchy` table, no?

>  Is there any solid reason to have mixed English, German, Polish and abbreviation translations in one big heap in the table?

You are probably talking about `gk-waypointy-type` table. All those are strings imported from all imports from other caching platforms. That were not standardized on initial import. So this was a nice and easy workaround to standardize them and have them translated.

kumy commented:
@okainov the project was retrieved ~ 2 years ago, it was stalled since a long time. I've cleaned it up a bit to remove some hard-coded password etc, some minor enhancements to the html (introduced bootstrap), so it was opensourced. I've migrated hosting to a fresh server and deployed website using docker containers. Recently, as I have to migrate again to a new host, I've scripted all the deployment process using ansible. We have worked hard to integrate build pipeline, and quality insurance tools thanks to travis, codacy and other tools. Migrated translation system to crowdin. We defined a release process and automatic deployments. We go gradually.

I've passed some time prototyping and building a JSON REST api over the actual database schema. We still didn't started big refactoring as the actual code is like spaghetti. Lot of code was and is still in polish (@boly38 and I are french, it's not easy). There is really huge amount of work still to be done, we're (I'm) quite afraid in refactor as it's really lots of work and code base is quite unstable. Each time something is started, it become a real huge amount of things to change. Manual testing don't help.

On his side, @boly38 has contributed a lot to many parts of the code, some refactor, introduced classes, QA tools etc. He worked on windows deployment, and many more things.

Last month, I've started to work actively on splitting html from code, and started to redesign the theme. I thought it was a work of some days, but it become more and more work as I advanced. All this work is in branch `new-theme` (which is continuously deployed to https://new-theme.staging.geokrety.org/). Since some days, I spend lot of time on user experience on logging page. This branch is now far away master and production branches. This branch still need to be squashed, as it's currently my "own dev branch". Thanks to this work, code start to be be cleaner and shorter.

I think we should spend our energy in finishing this work first, then we can do much more from this "next base". We need hands, and fresh blood, are you interested in helping us deeper? @okainov are you involved in other projects?

(We also need a Roadmap, documentations, contributors)

kumy commented:
`drop tables` are removed from the export.

okainov commented:
@kumy I'm mostly the only maintainer of geocaching.su and it has absolutely the same issues (legacy PHP, no tests, logic/templates mix and so on), so I can understand your pain =) But gc.su is not OpenSource (there is still a project owner/admin, even though he's more observer), so the code is not public (at least, yet). I'm trying to improve it from time to time, but it goes really slow, probably ATM it's in worse condition than GK. Apart of it I was contributing a bit into cgeo, but not much really. Actually I'm fulltime QA, so development is somehow part-time activity, mostly tools and scripting (that's why my code may be not very perfect -_-). 
I'd like to participate deeper where I can and do my best, but probably I won't have a lot of time, I also still try to travel around and do actual geocaching =) If it's ok from your side, then no problem ;)

kumy commented:
> contributing a bit into cgeo

hehe that's why I thought I "know" you :p (at least the gc.su OAuth integration ;))

> I'd like to participate deeper where I can and do my best, but probably I won't have a lot of time, I also still try to travel around and do actual geocaching =) If it's ok from your side, then no problem ;)

Sure, you're welcome to participate where you can, where you want, when you wish, to rephrase some @boly38 sentence, it's open source magic! So feel free to open issues and PR :) Contributions to `new-theme` branch are more than welcome starting next week (The time for me to clean this a bit and squash to a huge big commit :()

Good luck in opensourcing gc.su

okainov commented on docker/mariadb/00_database-create-geokrety-db.sql:
```diff
@@ -1,216 +1,155 @@
-  `name` varchar(128) COLLATE utf8_polish_ci NOT NULL
+  `name` varchar(128) COLLATE utf8_polish_ci NOT NULL,
+  UNIQUE KEY `groupid` (`groupid`),
+  UNIQUE KEY `groupid_2` (`groupid`)
```
> >Sounds like extra useless key :)
> 
> No context, I don't understand sorry.

Here

kumy replied:
Second unique key dropped.


BTW, I never seen the code using this table, probably some code never released.

okainov commented on docker/mariadb/00_database-create-geokrety-db.sql:
```diff
@@ -219,177 +158,146 @@ CREATE TABLE `gk-miasta` (
   `created` datetime NOT NULL COMMENT 'kiedy utworzono',
   `raceOwner` bigint(20) NOT NULL,
   `private` binary(1) NOT NULL DEFAULT '0' COMMENT '0 = public, 1 = private',
-  `haslo` varchar(16) COLLATE utf8_bin NOT NULL COMMENT 'haslo tajnego wyścigu - password for private course',
```
Here about comments, still nice to have English comments

kumy replied:
Fixed

okainov commented on docker/mariadb/00_database-create-geokrety-db.sql:
```diff
@@ -1,216 +1,155 @@
-  `zdjecia` smallint(5) UNSIGNED NOT NULL DEFAULT '0',
-  `ost_pozycja_id` int(10) UNSIGNED NOT NULL DEFAULT '0' COMMENT 'position last ruchy id',
-  `ost_log_id` int(10) UNSIGNED NOT NULL DEFAULT '0' COMMENT 'log last ruchy id',
+  `droga` int(10) unsigned NOT NULL,
```
Here also about comments, it was kinda nice and understandable...

kumy replied:
fixed

okainov commented on docker/mariadb/00_database-create-geokrety-db.sql:
```diff
@@ -400,59 +308,60 @@ CREATE TABLE `gk-races-krety` (
+  KEY `waypoint` (`waypoint`),
+  KEY `user` (`user`),
+  KEY `lat` (`lat`),
+  KEY `lon` (`lon`),
```
> I don't know the typical queries against this table, but as per amount of indexes looks like introducing some composite indexes may be helpful
> 
> Sounds like you're talking about gk-ruchy table, no?

Correct, this one

kumy replied:
In the `new-theme` branch, found sql queries on this table should be round up there https://github.com/geokrety/geokrety-website/blob/feature/new-theme/website/lib/geokrety/repository/TripRepository.class.php

Feel free to propose new indexes, I'll push them to new-theme (and/or staging branch) so we can test them.

FYI, this is the biggest table in the database, it actually contains 1713876 entries. So proper indexing would be nice. Database updates may take a certain amount of time ;)

kumy commented:
@okainov did you checked the translations on crowdin? The Russian website translation is only 58% ;)

https://crwd.in/geokrety

okainov commented:
@kumy  I wanted to, but link from the site didn't work, I mean,. it worked to add Crowdin interface directly to the website which was ugly and unusable

okainov commented:
@kumy ru translation 85+% now. But not sure is there any approver? Could you grant me approver permissions for ru maybe?

kumy commented:
@okainov Done, you're now proofreader for russian :+1:

kumy commented:
@boly38 may you check I'm not breaking `gkshell` please.
